src/crystal/project/git_dependency.cr

```diff
@@ -0,0 +1,60 @@
+require "crystal/project/dependency"
+require "crystal/project/project_error"
+
+module Crystal
+  class GitDependency < Dependency
+    getter target_dir
+
+    def initialize(@repo, name = nil : String?, branch = nil : String?)
+      unless repo =~ /(.*)(:|\/)(.*)\/(.*)(.git)/
```

Comment:
if we use syntax similar to bundler, then name will be there `gem "gem-name-here", git: "repo-url-here"`. In this case might be:

```crystal
deps do
  dep "frank", github: "manastech/frank"
  dep "super-framework", git: "git@super-secret.example.org:super-framework"
end
```